### ISSUE TITLE
Cannot access settings - Win7 64

### ISSUE THREAD

**Issue by DGmCart** [LMMS/lmms#1865]
LMMS crashes every time I try to access Edit > Settings. Otherwise it seems to be working fine. I've tried uninstalling and reinstalling but the problem persists. 

Possibly relevant - I had some problems initially with not getting any sound, as it turned out I just needed to update the driver for my soundcard. But before I figured that out I had tried running the 32bit version to see if that produced sound. I think that may have been the last time I successfully accessed the settings menu. After I switched to the 64bit version Edit > Settings causes instant death - "This application has requested that the runtime terminate it in an unusual way"

Any ideas?
(Windows 7 64, core i5 2540m, 4GB)

**Comment by tresf** [comment]
Possibly related #1128.

Can you open the run dialog (Windows Key + R) and paste the following:

```
cmd /k move %userprofile%\.lmmsrc.xml %userprofile%\.lmmsrc_bak.xml  & pause & exit
```

You should see:

![image](https://cloud.githubusercontent.com/assets/6345473/6631051/a7beda96-c8f6-11e4-86c0-180dbb1f5a0c.png)

If instead you see:


![image](https://cloud.githubusercontent.com/assets/6345473/6631071/f6e0fdfc-c8f6-11e4-9df2-48d3a510d62a.png)

Then something is wrong and you'll have to check the security/permissions of the file.

-Tres

**Comment by DGmCart** [comment]
The command worked as expected (1 file moved), but now LMMS crashes while loading. I just tried wiping it and reinstalling it again but it still crashes while loading. 

This time I got some error messages during the install:
Error opening file for writing:
C:\Program Files\LMMS\plugins\32\QtCore4.dll
Also for
C:\Program Files\LMMS\plugins\32\RemoteVstPlugin32.exe
C:\Program Files\LMMS\plugins\32\libwinpthread-1.dll
C:\Program Files\LMMS\plugins\32\zlib1.dll

No idea if this related or a separate problem.

**Comment by DGmCart** [comment]
This is the file. When you said send it to a dev, did you mean you or
someone else?

Thx - DG

**Comment by tresf** [comment]
I'm a good one to send it to, but GitHub strips out file attachments.  But let's hold off on emailing it just yet... 

The fact that it won't start now suggests you have a problem with your file permissions.

Can you create a second user profile on the machine, login, and verify LMMS starts just fine?  If so, you'll have to repair the permissions on your user profile to fix the problem you are encountering.  There are likely several tutorials on the internet for this and they would be unrelated to our software specifically. :+1: 

-Tres

**Comment by DGmCart** [comment]
Unfortunately this didn't work, LMMS crashes on loading in the new user profile

**Comment by tresf** [comment]
That sounds like a botchced install then.  On occasion a DLL gets locked and the uninstall doesn't fully complete (our installer isn't intelligent enough to handle this scenario).

These are the steps to fully uninstall a botched version:

1.  Uninstall from Control Panel
2.  Reboot
3.  Delete (or rename) the LMMS folder that is still remaining in Program Files (as well as Program Files (x86) if applicable).
4.  Reinstall

Please let us know if this helps.

**Comment by DGmCart** [comment]
There is no LMMS folder to delete after the uninstall, so I couldn't perform step 3. Is there is a way to restore the LMMS folder after the uninstall? I could then rename it before doing a fresh installation

**Comment by Umcaruje** [comment]
This might be a stupid question, but are you running the installer as administrator?

**Comment by DGmCart** [comment]
There are no stupid questions, it's entirely possible I've missed something obvious! 
I am running the installer as administrator

**Comment by Umcaruje** [comment]
Ok, just needed to check :)

Why not try to install it somewhere other than your program files folder e.g. somewhere in your user folder?

**Comment by DGmCart** [comment]
Good idea. Sorry to say it didn't work, It still crashes while loading

**Comment by Sti2nd** [comment]
If it gives you runtime error my guess is that this will work https://github.com/LMMS/lmms/issues/512#issuecomment-48825153

**Comment by tresf** [comment]
@DGmCart,

That is very odd.  What version are you using and where did it you download it from?

Assuming it's 1.1.3 and from our official site, there are a few other things you can try... 


Can you open the run dialog (Windows Key + R) and paste the following:
```cmd
devmgmt.msc
```

Now find your audio devices and disable any devices listed and relaunch the software.

![image](https://cloud.githubusercontent.com/assets/6345473/6647115/8a4fa3c2-c9a1-11e4-8c2c-2fdc103a2d87.png)


If this works correctly you'll have "Dummy Audio" show up in the Settings Dialog.

![image](https://cloud.githubusercontent.com/assets/6345473/6647135/9b6f4a68-c9a1-11e4-8f98-251159205e9f.png)


If this allows the software to start, then it is a sound driver issue and we can link you to the knowledge-base article we have for that.

-Tres

**Comment by DGmCart** [comment]
Success - After disabling the audio devices LMMS runs without crashing. Can you link me to the article on resolving sound driver issues?
Thanks for your help

**Comment by tresf** [comment]
It is part of our release notes... 

> Known issues with this release:

> **Windows**
>  * Software fails to load on some systems (Please see bug #1173, #512)

But the steps are simple enough... here they are:

 1. First, try [updating your device driver](http://windows.microsoft.com/en-us/windows/update-driver-hardware-isnt-working-properly#update-driver-hardware-isnt-working-properly=windows-7). 
  1. First try the latest driver from the manufacturer's website.
  1. In the case of LMMS, some have had better success completely removing the device driver and installing the driver from Windows Update.
 1. Second, try replacing `libportaudio-2.dll` with one from the [PortAudioSharp project](http://portaudiosharp.googlecode.com/).
  1. Download [PortAudio.dll](https://portaudiosharp.googlecode.com/files/PortAudio.dll) from the PortAudioSharp page
  1. Extract `PortAudioSharp.dll`
  1. Rename it to `libportaudio-2.dll`
  1. Move it to `C:\Program Files\LMMS` ***
  1. *** **Note**, these steps may only work with the 32-bit version of LMMS.

**Comment by tresf** [comment]
Duplicate of #1173

**Comment by musikBear** [comment]
@tresf 
in ref. to
http://lmms.io/forum/viewtopic.php?f=3&t=604
I have tested the dll in question.
I like it :+1: 

Master 14. jul by Dave winXPsp3
Test1 : SDL
Used old projects perfectly. I compared the quality (by ear) to an exported version, and i could not point to any difference.
Export: wave + ogg, default settings : ok
So in respect to SDL the dll passed.

Test2 PORT-AUDIO w. ASIO4ALL
Loads & Devices found correctly
Plays perfectly, with no hickups at all 
Sound quality exelent
Export: wave + ogg, default settings : ok
However-
This dll will still crash fatally with BSOD if something goes wrong, as has ALL and ANY libportaudio-2 dlls with ASIO selected.
That happens *always*!
(also mentioned in the forum thread)
MS like to scare their customers, so
![recoverfrom](https://cloud.githubusercontent.com/assets/6662066/11318303/3cd319e0-904c-11e5-8dd8-fdd4afc83dfe.png)
that is what follows :/
I think that events like this can make a user uncertain of lmms' safeness, but it is a windows issue, and not even sure if it happens on anything else than xp (anyone with win7 seen this BSOD at ASIO crash??)
So in conclusion: imo, this dll works very well. There is no difference when it comes to using SDL, so that option everyone still have, and the ASIO support looks fine.
So imo - include the dll.
Now someone with other OS (vista, 7,8,10) need to reproduce.
MAC & Linux uses something different -right?

**Comment by tresf** [comment]
@musikBear this thread is about workarounds for gaining access to the settings dialog when there is a specific driver conflict and my PortAudioSharp post was one suggest of many to help resolve this.  The ASIO4ALL feedback is off-topic.  Please move it to the forum thread where it belongs.

**Comment by musikBear** [comment]
@tresf so in #1173 ? It was just because you referenced *this* thread in the forum, that i wrote it here. 1173 is where PortAudio**Sharp** is suggested
